packages/components/src/Chips/hooks/UseInternalChip.tsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
```

Comment:
Nit: shouldn't the file name be `useInternalChip.tsx` instead of `UseInternalChip.tsx`?

Another thought on this is could this not be a utility function since the hook is only returning a callback that isn't ever changing since it is passed everything as a parameter to the callback?

Reply:
ah good catch. it should be lowercase.

I started out with it as a util, and then realized it needed to be a `.tsx` file in order for it to handle the `<Icon/>` we are returning.

for no real reason other than convention, I thought it was weird to have a `util.tsx` rather than `util.ts` though I do see we have a `Datalist.utils.tsx` which I assume is that way for the same reason

I don't feel strongly about it being a hook vs a util. there's certainly no need for it to be a hook.

Reply:
I think a hook is fine but I think we should at least make it so that callback isn't recreated on every re-render. I feel like that may cause issues in the future

Reply:
I'm not sure I agree

using `useCallback` isn't always going to improve performance, and React is pretty good at re-rendering. this computation isn't particularly expensive.

if there's a specific problem we want to solve with a `useCallback` I'm open to it, but I don't like using it "just in case"
https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
I was initially thinking of pulling the function out of the hook and returning it there. I think it is minor enough we can leave it.
Ideally we rename the file to `useInternalChip.tsx` and I can approve

Reply:
>I was initially thinking of pulling the function out of the hook and returning it there. I think it is minor enough we can leave it.

gotcha. the only thing with that is we need it in 2 places, single and multiselect, so having it be an unexported function in the file would mean we duplicate it. it is only 2 instances, so I'm not totally opposed to doing that.

alternatively, I can make it just a util/function that we call rather than a hook like you had originally asked about.

Reply:
renamed ✅
moved function outside of hook to avoid being remade ✅